src/new_hope/diagonal_difference.clj

```diff
@@ -0,0 +1,50 @@
+(ns new-hope.diagonal-difference
+  (:require [midje.sweet :refer :all]
+            [criterium.core :as crit]))
+
+;https://www.hackerrank.com/challenges/diagonal-difference
+
+(def matrix [[11 2 4] [4 5 6] [10 8 -12]])
+
+(defn element [m i j] (nth (nth m i) j))
+
+(defn first-sum [m, n]
```

Comment:
you can do instead
```clj
(defn diagonal1 [n]
  (for [i (range n)]
    [i i]))
(defn diagonal2 [n]
  (for [i (range n)]
    [i (- (dec n) i)]))
(facts "about diagonals"
  (diagonal1 3) => [[0 0] [1 1] [2 2]]
  (diagonal2 3) => [[0 2] [1 1] [2 0]])
```
and then define a generic function that sums all elements by their positions